shoes-core/spec/shoes/console_spec.rb

```diff
@@ -0,0 +1,215 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe Shoes::Console do
+  # include_context 'dsl app'
+
+  def call_button_mock_block(button_content)
+    button_content.gui.instance_variable_get(:@mock_clickable_block).call
+  end
+
+  let(:console) { Shoes.console }
+
+  let(:sample_message_array) { [[:debug, 'debug message'], [:warn, 'warn message'], [:error, 'error message']] }
+  let(:formatted_message_output){ "Debug\n debug message\n\nWarn\n warn message\n\nError\n error message\n\n" }
+
+  let(:mock_dialog) { Shoes::Mock::Dialog.new.dialog_chooser }
+
+  before(:each) do
+    # Depending on spec order, these can be leftover and contaminate other specs
+    console.instance_variable_set(:@app, nil)
+    console.instance_variable_set(:@messages, [])
+  end
+
+  describe '#initialize' do
+    it 'must reset @messages as []' do
+      console.instance_variable_set(:@messages, ['placeholder'])
+      console.send(:initialize)
+      expect(console.instance_variable_get(:@messages)).to eq([])
+    end
+  end
+
+  describe '#show' do
+    before(:each) do
+      allow(console).to receive(:create_app) { :create_app_called }
+      console.instance_variable_set(:@app, OpenStruct.new(focus: :app_focus_called))
+    end
+
+    context 'when #showing? is true' do
+      before(:each) { allow(console).to receive(:showing?) { true } }
+
+      it 'must call @app.focus' do
+        expect(console.show).to eq(:app_focus_called)
+      end
+    end
+
+    context 'when #showing? is false' do
+      before(:each) { allow(console).to receive(:showing?) { false } }
+
+      it 'must call @app.focus' do
+        expect(console.show).to eq(:create_app_called)
+      end
+    end
+  end
+
+  describe '#create app' do
+    let!(:console_app) do
+      console.instance_variable_set(:@messages, sample_message_array.dup)
+      console.create_app
+      console.instance_variable_get(:@app).instance_variable_get(:@__app__)
+    end
+
+    let!(:console_app_nested_contents) { console_app.contents.first.contents.first.contents }
+
+    context 'the contents ordering' do
```

Comment:
Honestly would be inclined to not test the specific ordering of this, since it introduces friction if we want to shuffle the sequence/nesting of things around in the UI to get the appearance we're after. Thank you for being thorough, though!